lib/debezium/schema.go

```diff
@@ -81,70 +81,69 @@ func (f Field) GetScaleAndPrecision() (int32, *int32, error) {
+func (f Field) ToValueConverter() (converters.ValueConverter, error) {
 	switch f.DebeziumType {
 	// Passthrough converters
 	case UUID, LTree, Enum:
-		return converters.StringPassthrough{}
+		return converters.StringPassthrough{}, nil
 	case Year, MicroDuration:
-		return &converters.Int64Passthrough{}
+		return &converters.Int64Passthrough{}, nil
 	case DateTimeWithTimezone:
-		return converters.DateTimeWithTimezone{}
+		return converters.DateTimeWithTimezone{}, nil
 	case TimeWithTimezone:
-		return converters.TimeWithTimezone{}
+		return converters.TimeWithTimezone{}, nil
 	case GeometryPointType:
-		return converters.GeometryPoint{}
+		return converters.GeometryPoint{}, nil
 	case GeographyType, GeometryType:
-		return converters.Geometry{}
+		return converters.Geometry{}, nil
 	case JSON:
-		return converters.JSON{}
+		return converters.JSON{}, nil
 	case Date, DateKafkaConnect:
-		return converters.Date{}
+		return converters.Date{}, nil
 	// Decimal
 	case KafkaVariableNumericType:
-		return converters.NewVariableDecimal()
+		return converters.NewVariableDecimal(), nil
+	case KafkaDecimalType:
+		scale, precisionPtr, err := f.GetScaleAndPrecision()
+		if err != nil {
+			return nil, err
+		}
+
+		precision := decimal.PrecisionNotSpecified
+		if precisionPtr != nil {
+			precision = *precisionPtr
+		}
+
+		return converters.NewDecimal(decimal.NewDetails(precision, scale)), nil
 	// Time
 	case Time, TimeKafkaConnect:
-		return converters.Time{}
+		return converters.Time{}, nil
 	case NanoTime:
-		return converters.NanoTime{}
+		return converters.NanoTime{}, nil
 	case MicroTime:
-		return converters.MicroTime{}
+		return converters.MicroTime{}, nil
 	// Timestamp
 	case Timestamp, TimestampKafkaConnect:
-		return converters.Timestamp{}
+		return converters.Timestamp{}, nil
 	case MicroTimestamp:
-		return converters.MicroTimestamp{}
+		return converters.MicroTimestamp{}, nil
 	case NanoTimestamp:
-		return converters.NanoTimestamp{}
+		return converters.NanoTimestamp{}, nil
 	}
 
-	return nil
+	return nil, nil
 }
 
 func (f Field) ToKindDetails() typing.KindDetails {
 	// Prioritize converters
-	if converter := f.ToValueConverter(); converter != nil {
-		return converter.ToKindDetails()
+	converter, err := f.ToValueConverter()
+	if err != nil {
+		// TODO: Return an actual error instead.
+		return typing.Invalid
 	}
 
-	// TODO: Deprecate this in favor of the converters
-
-	// We'll first cast based on Debezium types
-	// Then, we'll fall back on the actual data types.
-	switch f.DebeziumType {
-	case KafkaDecimalType:
-		scale, precisionPtr, err := f.GetScaleAndPrecision()
-		if err != nil {
-			return typing.Invalid
-		}
-
-		precision := decimal.PrecisionNotSpecified
-		if precisionPtr != nil {
-			precision = *precisionPtr
-		}
-
-		return typing.NewDecimalDetailsFromTemplate(typing.EDecimal, decimal.NewDetails(precision, scale))
+	if converter != nil {
+		return converter.ToKindDetails()
 	}
 
 	if f.DebeziumType != "" {
```

Comment:
Could move this to the `default` case of the switch statement in `ToValueConverter()`.

Reply:
sgtm